avocado/core/plugin_interfaces.py

```diff
@@ -266,9 +266,34 @@ class Spawner(Plugin):
 
     @staticmethod
     @abc.abstractmethod
-    def is_task_alive(task):
-        """Determines if a task is alive or not."""
+    def is_task_alive(task_info):
+        """Determines if a task is alive or not.
+
+        :param task_info: wrapper for a Task with additional runtime information
+        :type task_info: :class:`avocado.core.task.info.TaskInfo`
+        """
 
     @abc.abstractmethod
-    async def spawn_task(self, task):
-        """Spawns a task return whether the spawning was successful."""
+    async def spawn_task(self, task_info):
+        """Spawns a task return whether the spawning was successful.
+
+        :param task_info: wrapper for a Task with additional runtime information
+        :type task_info: :class:`avocado.core.task.info.TaskInfo`
+        """
+
+    @abc.abstractmethod
+    async def wait_task(self, task_info):
+        """Waits for a task to finish.
+
+        :param task_info: wrapper for a Task with additional runtime information
+        :type task_info: :class:`avocado.core.task.info.TaskInfo`
+        """
+
+    @staticmethod
+    @abc.abstractmethod
+    async def check_task_requirements(task_info):
+        """Checks if the requirements described within a task are available.
+
+        :param task_info: wrapper for a Task with additional runtime information
+        :type task_info: :class:`avocado.core.task.info.TaskInfo`
+        """
```

Comment:
Hi @clebergnu, how complete is the API at this stage? Do we have any other required methods for a spawner plugin specified in any of the latest blueprints?

Reply:
Hi @pevogam, I expect that this API (as in the end of this PR) will be the final shape for releases 81.0 and 82.0.  Unfortunately, none of the blueprints touched on the subject of spawners directly.  After 82.0, we're free to expand/improve this API, so I'd love to hear your ideas.

Reply:
That's great news then! Yes, I think I have some ideas that I want to try out but was waiting for a green flag like this one. :green_circle: